astro/src/content/docs/lifecycle/migrate-users/bulk/stytch.mdx

```diff
@@ -0,0 +1,327 @@
+
+<Aside type="note">
+The `1_makeStytchUsers.mjs` file uses the Stytch JavaScript SDK to create users with three different hashing algorithms. However, the line `client.passwords.authenticate` provides the cleartext password to Stytch, which Stytch uses to rehash the user password using the Scrypt hashing algorithm. At the end of this script, all your users' passwords will be hashed with Scrypt in the Stytch database.
+
+Even though you can create passwords for Stytch users with different hashing algorithms ([bcrypt, scrypt, Argon2, MD5, SHA-1, or PBKDF2](https://stytch.com/docs/api/password-migrate)), scrypt is the Stytch-preferred algorithm. Whenever a user logs in, Stytch will verify their password hash using the algorithm it is currently stored with, and then rehash their password using scrypt and discard the old hash. Password rehashing is a common technique used to upgrade security as hashing algorithms evolve.
+
+Once you have completed this tutorial and understand how to import hashes into FusionAuth, you might want to go back and comment out the `authenticate` line to test other hashing algorithms. Note that this is necessary only in the rare case that you previously imported users into Stytch from another system, and some of those users have not yet logged in again.
+</Aside>
+
+## Request User Passwords From Stytch
+
+You cannot download your users' password hashes from Stytch using their API. To get them, email support@stytch.com from the email address you used to sign up with Stytch. Ask support to send your users' hashes to you.
+
+### Email The Request
+
+Stytch will encrypt the hashes with your public key. Attach this key in a `.pem` file to the email you send Stytch.
+
+To create a private and public key pair, use the commands below in a terminal.
+
+```shell
+openssl genpkey -algorithm RSA -out private_key.pem &&
+openssl rsa -pubout -in private_key.pem -out public_key.pem
+```
+
+Email Stytch only `public_key.pem`. Keep `private_key.pem` secret and secure.
+
+An example of what these keys look like is in the directory `fusionauth-import-scripts/stytch/exampleData/1_emailRequest`. You can use the keys to request test users from Stytch but do **not** use these keys for real users, as they are publicly available on GitHub.
+
+<Aside type="tip">
+If you don't want to send Stytch an email, you can use the example response files for the rest of this tutorial. The two files Stytch emailed are in `exampleData/2_emailResponse`. Just remember to change the sample user Ids to your actual user Ids as you proceed.
+</Aside>
+
+### Decrypt The Reply
+
+Stytch will reply with two files: An encrypted password hash file (`stytch-project-test-36510638-652a-4d3d-9a94-f0a7106582fc-hashes-2021-01-11.enc`) and the key to decrypt it (`key.bin.enc`).
+
+Decrypt the password file with the commands below.
+
+```shell
```

Comment:
most of these code snippets have the same leading space on the first line issue.